tools/codegen/operator_versions/gen_mobile_upgraders.py

```diff
@@ -0,0 +1,285 @@
+import sys
+import os
+import yaml
+from tools.codegen.code_template import CodeTemplate
+from typing import Dict, List
+from enum import Enum
+
+class ByteCode(Enum):
+    instructions = 1
+    constants = 2
+    types = 3
+    operators = 4
+    register_size = 5
+
+def load_yaml(upgrader_yaml_path):
+    with open(upgrader_yaml_path, "rb") as yaml_file:
+        return yaml.safe_load(yaml_file)
+
+
+ONE_INSTRUCTION = CodeTemplate("""
+    Instruction{OpCode::${operator_name}, ${X}, ${N}},""")
+
+INSTRUCTION_LIST = CodeTemplate("""std::vector<Instruction>({
+        ${instruction_list}
+    }), // instructions list""")
+
+ONE_CONSTANT = CodeTemplate("""c10::IValue(${constant}),""")
+
+CONSTANT_LIST = CodeTemplate("""std::vector<c10::IValue>({
+        ${constant_list}
+    }), // constants list""")
+
+ONE_TYPE = CodeTemplate("""c10::parseType("${type_str}"),""")
+
+TYPE_LIST = CodeTemplate("""std::vector<c10::TypePtr>({
+        ${type_list}
+    }), // types list""")
+
+ONE_OPERATOTR_STRING = CodeTemplate("""
+    OperatorString({"${operator_name}", "${overload_name}", ${num_of_args}}),""")
+
+OPERATOR_STRING_LIST = CodeTemplate("""
+    std::vector<OperatorString>({
+        ${operator_string_list}
+    }), // operators list""")
+
+ONE_UPGRADER_FUNCTION = CodeTemplate("""
+    mobile::Function::registerFunc(
+        "${upgrader_name}",
+        ${instruction_list},
+        ${constant_list},
+        ${type_list},
+        ${register_size}
+    )""")
+
+ONE_UPGRADER_SRC = CodeTemplate("""
+    ByteCodeFunctionWithOperator({
+        ${bytecode_function},
+        ${operator_string_list}
+    }),""")
+
+
+UPGRADER_CPP_SRC = CodeTemplate("""#pragma once
+/**
+ * Generated by gen_mobile_upgraders.py
+ */
+
+#include <ATen/core/ivalue_inl.h>
+#include <torch/csrc/jit/mobile/code.h>
+#include <torch/csrc/jit/mobile/function.h>
+#include <torch/csrc/jit/serialization/import_export_functions.h>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace c10 {
+TypePtr parseType(const std::string& pythonStr);
+} // namespace c10
+
+namespace torch {
+namespace jit {
+
+struct Instruction;
+struct Upgrader {
+  int min_version;
+  int max_version;
+  std::string upgrader_name;
+  int index;
+};
+
+// From operator_versions.yaml
+const std::unordered_map<std::string, std::vector<Upgrader>> kOperatorVersionMap(
+    {
+        {std::string("aten::div.Tensor"),
+         std::vector<Upgrader>({Upgrader({0, 3, "div_Tensor_0_3", 0})})},
+        {std::string("aten::div.Scalar"),
+         std::vector<Upgrader>({Upgrader({0, 3, "div_Scalar_0_3", 1})})},
+        {std::string("aten::div.out"),
+         std::vector<Upgrader>({Upgrader({0, 3, "div_out_0_3", 2})})},
+        {std::string("aten::div_.Tensor"),
+         std::vector<Upgrader>({Upgrader({0, 3, "div__Tensor_0_3", 3})})},
+        {std::string("aten::div_.Scalar"),
+         std::vector<Upgrader>({Upgrader({0, 3, "div__Scalar_0_3", 4})})},
+    });
```

Comment:
Why are upgrader definitions hard coded here?
@tugsbayasgalan I think you mentioned that operator_versions.yaml will be deleted, then what will replace it? It seems we are still referencing it here.

Reply:
Yeah this python will take an input bytecode yaml and output cpp. 

In the latest commit, `kOperatorVersionMap` is generated from `torch._C._get_operator_version_map()`, which is populated from `caffe2/torch/csrc/jit/operator_upgraders/version_map.h`  in https://github.com/pytorch/pytorch/pull/68772.

Reply:
In https://github.com/pytorch/pytorch/pull/68005, `def generate_bytecode(file_name)` will generate a bytecode yaml file, I will append this function at the end of `generate_bytecode` to generate a bytecode cpp instead.

Reply:
The reason why codegen another version map is that it needs to map the the index in `std::vector<ByteCodeFunctionWithOperator> kUpgraderByteCode`. When replacing the operator in bytecode
`OP`, `X`, `N`, we can replace `OP` with `CALL` and `X` with the according index in `kUpgraderByteCode`. 

Maybe there is a more convenient way to replace. What I describe above is the implementation in https://github.com/pytorch/pytorch/pull/67731